File: prompts/fixtures/ts/debug_test.ts
"use strict";

function add(a: number, b: number): number {
    const result: number = a + b;      // line 4
    return result;                      // line 5
}

function multiply(a: number, b: number): number {
    const result: number = a * b;      // line 9
    return result;                      // line 10
}

function compute(x: number, y: number): number {
    const sum: number = add(x, y);             // line 14
    const product: number = multiply(x, y);    // line 15
    const final_: number = sum + product;      // line 16
    return final_;                              // line 17
}

function loopSum(n: number): number {
    let total: number = 0;                      // line 21
    for (let i: number = 1; i <= n; i++) {
        total = add(total, i);                 // line 23
    }
    return total;                               // line 25
}

function factorial(n: number): number {
    if (n <= 1) return 1;                      // line 29
    return n * factorial(n - 1);               // line 30
}

function main(): void {
    const x: number = 10;                               // line 34
    const y: number = 20;                               // line 35
    const result1: number = compute(x, y);              // line 36
    console.log(`compute = ${result1}`);                // line 37
    const result2: number = loopSum(5);                 // line 38
    console.log(`loop_sum = ${result2}`);               // line 39
    const result3: number = factorial(5);               // line 40
    console.log(`fact = ${result3}`);                   // line 41
}

main();

File: prompts/fixtures/java/DebugTest.java
public class DebugTest {
    static int add(int a, int b) {
        int result = a + b;                           // line 3
        return result;                                // line 4
    }

    static int multiply(int a, int b) {
        int result = a * b;                           // line 8
        return result;                                // line 9
    }

    static int compute(int x, int y) {
        int sum = add(x, y);                          // line 13
        int product = multiply(x, y);                 // line 14
        int fin = sum + product;                      // line 15
        return fin;                                   // line 16
    }

    static int loopSum(int n) {
        int total = 0;                                // line 20
        for (int i = 1; i <= n; i++) {
            total = add(total, i);                    // line 22
        }
        return total;                                 // line 24
    }

    static int factorial(int n) {
        if (n <= 1) return 1;                         // line 28
        return n * factorial(n - 1);                  // line 29
    }

    public static void main(String[] args) {
        int x = 10;                                   // line 33
        int y = 20;                                   // line 34
        int result1 = compute(x, y);                  // line 35
        System.out.printf("compute = %d%n", result1); // line 36
        int result2 = loopSum(5);                     // line 37
        System.out.printf("loop_sum = %d%n", result2);// line 38
        int result3 = factorial(5);                   // line 39
        System.out.printf("fact = %d%n", result3);    // line 40
    }
}
